File: src/main/java/com/gromart/springboot/service/OrderService.java
package com.gromart.springboot.service;

import com.gromart.springboot.model.Order;

import java.util.Map;

public interface OrderService {
    Map<String, Object> findAllOrders(int page, int limit);

    void saveOrder (Order order);

    Map <String, Object> findByUserId (String userId, int page, int limit);

    Map<String, Object> findByOrderDate (String startDate, String toDate, int page, int limit);

    Map<String, Object> findByOrderId (String orderId, int page, int limit);

    Map<String, Object> findByStatus(Boolean status, int page, int limit);

    Order findById (String orderId);

    void deleteOrderById(String orderId);

    int countDetail (String cartId);

    void updateStatus (Order order);

    Map<String, Object> countTransaction ();




}

File: src/service/OrderService.js
import axios from "axios";

const ORDER_API = "http://localhost:8080/gromart/order/";

class OrderService {
  getOrderByUserID(userId, page, limit) {
    return axios.get(
      ORDER_API + "id/" + userId + "/?page=" + page + "&limit=" + limit
    );
  }

  getAllOrder(page, limit) {
    return axios.get(ORDER_API + "/?page=" + page + "&limit=" + limit);
  }

  checkoutOrder(checkout) {
    return axios.post(ORDER_API, checkout);
  }

  getOrderById(orderId) {
    return axios.get(ORDER_API + orderId);
  }

  getTopSales() {
    return axios.get(ORDER_API + "topsales/");
  }

  getTotalTransaction() {
    return axios.get(ORDER_API + "count/");
  }

  updateStatus(update) {
    return axios.put(ORDER_API + "status/", update);
  }

  searchByOrderId(orderId, page, limit) {
    return axios.get(
      ORDER_API + "findid/" + orderId + "/?page=" + page + "&limit=" + limit
    );
  }

  searchByStatus(status, page, limit) {
    return axios.get(
      ORDER_API +
        "findStatus/?status=" +
        status +
        "&page=" +
        page +
        "&limit=" +
        limit
    );
  }

  searchByDate(startDate, toDate, page, limit) {
    return axios.get(
      ORDER_API +
        "date/?startDate=" +
        startDate +
        "&toDate=" +
        toDate +
        "&page=" +
        page +
        "&limit=" +
        limit
    );
  }
}

export default new OrderService();
